src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -87,11 +87,34 @@ protected AssetOccupancyProviderBase(IServiceProvider serviceProvider)
     }
 
     [UnitOfWork]
-    public virtual Task<List<PeriodOccupancyModel>> GetPeriodsAsync(AssetCategory category,
+    public virtual async Task<List<PeriodOccupancyModel>> GetPeriodsAsync(AssetCategory category,
         DateTime targetDate,
         DateTime? currentDateTime = default)
     {
-        throw new NotImplementedException();
+        var assets =
+            await AssetRepository.GetListAsync(x => x.AssetCategoryId == category.Id && !x.Disabled);
+
+        var result = new List<PeriodOccupancyModel>();
+        foreach (var asset in assets)
+        {
+            var periodOccupancyModels = await GetPeriodsAsync(asset, category, targetDate, currentDateTime);
+
+            foreach (var periodOccupancyModel in periodOccupancyModels)
+            {
+                var existModel = result.FirstOrDefault(x => x.PeriodId == periodOccupancyModel.PeriodId);
+                if (existModel is null)
+                {
+                    result.Add(periodOccupancyModel);
+                }
+                else
+                {
+                    existModel.TotalVolume += periodOccupancyModel.TotalVolume;
```

Comment:
I don't know why we sum the volume of all the assets. My understanding of booking by category is to book one of the assets in the specified category.

Reply:
> booking by category is to book one of the assets in the specified category

It's right, but search by category should return every asset's volume info or summary of all asset's volume info?

Reply:
Don't think complicated. Try to order the assets by `Priority` and choose the first available one.

Reply:
> Try to order the assets by `Priority` and choose the first available one.

Yeah, it's make sense.

Reply:
What if there is no available asset, we should return empty `PeriodOccupancyModel` list?

Reply:
Only `Asset.PeriodId == Category.PeriodId` are included in calculating. It will not be empty since you will get a result based on the period scheme.